### DISCUSSION TITLE
prefill form field data is showing but validation is not working

### DISCUSSION THREAD

**Discussion by deepakchandrawanshi1063** [react-hook-form/react-hook-form#6689] [Q&A] [answered]
const data={address:
  {  district: "Ranchi"
    landmark: "shiv temple"
    pincode: 834002
    postoffice: "hinoo"
    state: "Jharkhand"
    street: "saket nagar"}
}

//useForm
 const {
    register:register4,
    handleSubmit:handleSubmit4,
    reset:reset4,
    formState: { errors:errors4 },
  } = useForm({defaultValues:data});

//render (getting error msg:TypeError: Cannot read properties of undefined (reading 'landmark') in errors4.address.landmark)
<label htmlFor="landmark">Landmark</label>
                        <input className="form-control" type="text" {...register4("address.landmark",{
                           pattern: {
                              value: /^[a-zA-Z0-9\s,'-,:]{3,50}$/,
                              message: "Enter 3-50 Characters",
                            },
                            required:true,
                          })}/>
                          <br/>
                        {errors4.address.landmark && (
                          <span style={{ color: "red" }}>
                            {errors4.address.landmark.message}
                          </span>
                        )}
//

**Comment by thanh-nguyen-95** [comment, marked as answer]
I don't think validation is ran on initial render, which makes `formState.errors.address` undefined. You can try calling [`trigger()`](https://react-hook-form.com/api/useform/trigger) after the initial render to validate the form.

Make sure you check if the error exists first by using the `?.` syntax like so:

```jsx
<span style={{ color: 'red' }}>
  {errors4.address?.landmark?.message}
</span>
```

**Reply by bluebill1049** [reply]
That's right. we don't trigger validation right away, do in useEffect with trigger is the wise option.

**Reply by deepakchandrawanshi1063** [reply]
thank you all for solving this issue.
